src/storage-controller/src/lib.rs

```diff
@@ -933,46 +929,34 @@ where
+                                .extend_previous_statuses(last_status_per_id.into_iter().flatten())
                         }
                         IntrospectionType::SinkStatusHistory => {
-                            self.partially_truncate_status_history(
-                                IntrospectionType::SinkStatusHistory,
-                            )
-                            .await;
+                            let last_status_per_id = self
+                                .partially_truncate_status_history(
+                                    IntrospectionType::SinkStatusHistory,
+                                )
+                                .await;
+
+                            self.collection_status_manager
+                                .extend_previous_statuses(last_status_per_id.into_iter().flatten())
                         }
                         IntrospectionType::PrivatelinkConnectionStatusHistory => {
-                            // Truncate the private link connection status history table and
-                            // store the latest timestamp for each id in the table.
-                            let occurred_at_col =
-                                healthcheck::MZ_PRIVATELINK_CONNECTION_STATUS_HISTORY_DESC
-                                    .get_by_name(&ColumnName::from("occurred_at"))
-                                    .expect("schema has not changed")
-                                    .0;
-                            self.privatelink_status_table_latest = self
+                            let last_status_per_id = self
                                 .partially_truncate_status_history(
                                     IntrospectionType::PrivatelinkConnectionStatusHistory,
                                 )
-                                .await
-                                .and_then(|map| {
-                                    Some(
-                                        map.into_iter()
-                                            .map(|(id, row)| {
-                                                (
-                                                    id,
-                                                    row.iter()
-                                                        .nth(occurred_at_col)
-                                                        .expect("schema has not changed")
-                                                        .unwrap_timestamptz()
-                                                        .into(),
-                                                )
-                                            })
-                                            .collect(),
-                                    )
-                                });
+                                .await;
+
+                            self.collection_status_manager
+                                .extend_previous_statuses(last_status_per_id.into_iter().flatten())
```

Comment:
the privatelink statuses aren't currently handled by the `collection_status_manager` so I don't think there is any benefit to doing this

Reply:
Agreed--mentioned above but will unify these or revert.